modules/gapi/misc/python/pyopencv_gapi.hpp

```diff
@@ -124,6 +125,64 @@ PyObject* pyopencv_from(const cv::detail::PyObjectHolder& v)
     return o;
 }
 
+// #FIXME: Is it possible to implement pyopencv_from/pyopencv_to for generic
+// cv::variant<Types...> ?
+template <>
+PyObject* pyopencv_from(const cv::gapi::wip::draw::Prim& prim)
+{
+    switch (prim.index()) {
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Rect>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Rect>(prim));
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Text>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Text>(prim));
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Circle>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Circle>(prim));
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Line>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Line>(prim));
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Poly>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Poly>(prim));
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Mosaic>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Mosaic>(prim));
+        case cv::gapi::wip::draw::Prim::index_of<cv::gapi::wip::draw::Image>():
+            return pyopencv_from(cv::util::get<cv::gapi::wip::draw::Image>(prim));
+    }
+
+    util::throw_error(std::logic_error("Unsupported draw primitive type"));
+}
+
+template <>
+PyObject* pyopencv_from(const cv::gapi::wip::draw::Prims& value)
+{
+    return pyopencv_from_generic_vec(value);
+}
+
+template<>
+bool pyopencv_to(PyObject* obj, cv::gapi::wip::draw::Prim& value, const ArgInfo& info)
+{
+#define TRY_EXTRACT(Prim)                                                                                  \
+    if (PyObject_TypeCheck(obj, reinterpret_cast<PyTypeObject*>(pyopencv_gapi_wip_draw_##Prim##_TypePtr))) \
```

Comment:
@alalek Is there any way to implemente it via more high level functions ? `pyopencv_to`/`pyopencv_to_safe` aren't suitable there, are they ? Because they set an `error` in case mismatch.

Reply:
Not sure that do you mean here.
"dynamic types" are handled though switches/virtual functions.

If you don't want to follow `pyopencv_to/pyopencv_from` conventions, then please change names of these functions.

Reply:
What do u mean by "dynamic types". Is the `cv::variant<Types...>` is a dynamic type ? 
It seems to me since `cv::variant<>` has the limited number of possible types it can be wrapped like this:
```
cv::gapi::wip::draw::Rect rect;
if (pyopencv_to(obj, rect, info))
{
    value = rect;
    return true;
}
...
if(pyopencv_to(obj, text, info)
{
    value = text;
    return false;
}

// set error
return false;
```

But it doesn't work because in case fail first `pyopencv_to` setup an error.

Is it possible to implement `pyopencv_to` for `variant<>` following all conventions somehow else it's done above ?

Reply:
Variant implementation always has a `tag`/`kind` field which says which underlying object is stored.
There is NO sequence of calls.
There is `switch(tag)` with ONE call.

So you need to determine what is the "tag" value in that case. Perhaps switching over `PyObject_TypeCheck` is enough.

Reply:
But in this case `variant<>` is empty, isn't it ? 
It seems I can't just go through `switch` to determinate the type, cuz there is no stored object so far

Reply:
I've implemented `pyopencv_from` like you proposed above